Feature/category

micaelpimentel opened the pull request:
Implemented Category class

micaelpimentel committed:
Added Alternative class

micaelpimentel committed:
teste with alternative objects

micaelpimentel committed:
Added category class

MathBrandino commented on src/main/kotlin/Category.kt:
```diff
@@ -0,0 +1,25 @@
+            throw IllegalArgumentException("Código não pode ser vazio")
+        if (!code.matches("^(?:[a-z]-?)+[^-]\$".toRegex()))
+            throw IllegalArgumentException("Código deve ser composto por letras minusculas separadas por -")
+        if (!colorCode.matches("[a-fA-F0-9]{6}".toRegex()))
```
precisa começar com # pra ser um hexadecimal válido, mas sem crise deixar assim :)

micaelpimentel committed:
Adjusted regex

Now regex to match hex color code will look for a string starting with a
hash (#).
